### PR TITLE
Add Kubernetes Auth Method

### PR DESCRIPTION
**Pull request by hreeder**
Stackstorm-ha, by default, runs in Kubernetes (which is how I've got ST2 deployed). This PR should add support for Vault's Kubernetes auth method to the pack.

Draft, while I test it locally on my own deployment, but to give you the heads up this is coming.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,14 +13,15 @@ It should contain:
 * `cert` - Path to client-side certificate
 * `verify` - Whether to verify the SSL certificate or not
 * `auth_method` - Which authentication method to use.
-  Only `token` (the default) and `approle` are implemented so far.
+  Only `token` (the default), `approle` and `kubernetes` are implemented so far.
```
Comment (nzlosh) [inline README.md]: I'd recommend changing the wording to be neutral:
> Available implementations are: `token` (default), `approle` and `kubernetes`

**Comment by cognifloyd** [comment]
Oooh. This is pretty cool! Thanks!
`config.schema.yaml` will need to be updated to add `role` and `auth_method=kubernetes`.

File: README.md (hunk 1 above)
Comment (hreeder) [reply README.md]: Sounds good, changed 👍

**Comment by hreeder** [comment]
> Oooh. This is pretty cool! Thanks!
> `config.schema.yaml` will need to be updated to add `role` and `auth_method=kubernetes`.

Thanks for the catch, updated.

**Comment by hreeder** [comment]
@cognifloyd This should be good for review now. It's working in my environments suitably and we can access and retrieve vault secrets as necessary within our K8s environments.

I've also added a generic `auth_mount_point` config parameter as our setup makes use of custom mount points for different environments.

**Comment by cognifloyd** [comment]
#23 should fix the test errors here.

**Comment by cognifloyd** [comment]
#23 was just merged. Please merge in master. That should fix the tests.

**Comment by nzlosh** [comment]
Closing as stale.  Reopen with signed CLA and rebased master to have this PR merged.

**Pull request closed without merging**